### PR TITLE
Options update

### PR DESCRIPTION
**Pull request by Vsevolod-Livinskij**

### CODE REVIEW

**Comment by dbabokin** [comment]
- we also need hash of the generator itself, i.e. version string of the generator.
- options need to be in the form of command line options which can be copy-pasted to reproduce the results.

File: src/options.cpp [hunk 1]
```diff
@@ -421,3 +408,24 @@ void OptionParser::parseEmitPragmas(std::string val) {
...
+    Options &options = Options::getInstance();
+    options.setOutDir(std::move(val));
+}
+
+void Options::dump(std::ostream &stream) {
+    dumpVersion(stream);
+    stream << "Seed: " << seed << "\n";
+    stream << "Invocation: ";
+    for (const auto &option : raw_options) {
+        stream << option << " ";
```
Comment (dbabokin) [inline src/options.cpp:422]: Avoid trailing white space - print it before the argument, not after.

**Comment by dbabokin** [comment]
Fixes #118

File: src/options.cpp (hunk 1 above)
Comment (Vsevolod-Livinskij) [reply src/options.cpp]: Done

**Pull request merged**